packages/clearpay-integration/src/clearpay-payment-strategy.ts

```diff
@@ -0,0 +1,159 @@
+import { noop } from 'lodash';
+
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+    OrderFinalizationNotCompletedError,
+    OrderRequestBody,
+    PaymentArgumentInvalidError,
+    PaymentInitializeOptions,
+    PaymentIntegrationSelectors,
+    PaymentIntegrationService,
+    PaymentMethod,
+    PaymentRequestOptions,
+    PaymentStrategy,
+    RequestError,
+    RequestOptions,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import ClearpayScriptLoader from './clearpay-script-loader';
+import ClearpaySdk from './clearpay-sdk';
+
+export default class ClearpayPaymentStrategy implements PaymentStrategy {
+    private _clearpaySdk?: ClearpaySdk;
+
+    constructor(
+        private _paymentIntegrationService: PaymentIntegrationService,
+        private _clearpayScriptLoader: ClearpayScriptLoader,
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<void> {
+        const { getPaymentMethodOrThrow } = this._paymentIntegrationService.getState();
+        const paymentMethod = getPaymentMethodOrThrow(options.methodId, options.gatewayId);
+
+        this._clearpaySdk = await this._clearpayScriptLoader.load(paymentMethod);
+    }
+
+    deinitialize(): Promise<void> {
+        this._clearpaySdk = undefined;
+
+        return Promise.resolve();
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<void> {
+        if (!payload.payment) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
+        }
+
+        const { gatewayId, methodId } = payload.payment;
+
+        if (!gatewayId || !methodId) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
+        }
+
+        const { isStoreCreditApplied: useStoreCredit } = this._paymentIntegrationService
+            .getState()
+            .getCheckoutOrThrow();
+
+        await this._paymentIntegrationService.applyStoreCredit(useStoreCredit);
+
+        await this._paymentIntegrationService.validateCheckout(
+            this._paymentIntegrationService.getState().getCheckout(),
+            options,
+        );
+
+        const { countryCode } = this._paymentIntegrationService
+            .getState()
+            .getBillingAddressOrThrow();
+
+        if (!this._isCountrySupported(countryCode)) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because billing country is not supported.',
+            );
+        }
+
+        await this._loadPaymentMethod(gatewayId, methodId, options);
+
+        await this._redirectToClearpay(
+            countryCode,
+            this._paymentIntegrationService.getState().getPaymentMethod(methodId, gatewayId),
+        );
```

Comment:
To avoid this._paymentIntegrationService.getState() duplication it might be better to create state variable and use methods from there

```suggestion
        const state = this._paymentIntegrationService.getState();

        const { isStoreCreditApplied: useStoreCredit } = state.getCheckoutOrThrow();

        await this._paymentIntegrationService.applyStoreCredit(useStoreCredit);

        await this._paymentIntegrationService.validateCheckout(
            state.getCheckout(),
            options,
        );

        const { countryCode } = state.getBillingAddressOrThrow();

        if (!this._isCountrySupported(countryCode)) {
            throw new InvalidArgumentError(
                'Unable to proceed because billing country is not supported.',
            );
        }

        await this._loadPaymentMethod(gatewayId, methodId, options);

        await this._redirectToClearpay(
            countryCode,
            state.getPaymentMethod(methodId, gatewayId),
        );
```